packages/test/test-end-to-end-tests/src/test/compression.spec.ts

```diff
@@ -3,60 +3,117 @@
  * Licensed under the MIT License.
  */
 
+// eslint-disable-next-line import/no-nodejs-modules
+import * as crypto from "crypto";
 import { strict as assert } from "assert";
-import { Container } from "@fluidframework/container-loader";
-import { SharedMap } from "@fluidframework/map";
+import { ISharedMap, SharedMap } from "@fluidframework/map";
 import { requestFluidObject } from "@fluidframework/runtime-utils";
 import {
 	DataObjectFactoryType,
 	ITestContainerConfig,
 	ITestFluidObject,
 	ITestObjectProvider,
 } from "@fluidframework/test-utils";
-import { describeFullCompat } from "@fluidframework/test-version-utils";
+import {
+	describeFullCompat,
+	describeInstallVersions,
+	getVersionedTestObjectProvider,
+} from "@fluidframework/test-version-utils";
 import { CompressionAlgorithms } from "@fluidframework/container-runtime";
+import { pkgVersion } from "../packageVersion";
 
-const testContainerConfig: ITestContainerConfig = {
-	registry: [["mapKey", SharedMap.getFactory()]],
-	runtimeOptions: {
-		compressionOptions: {
-			minimumBatchSizeInBytes: 1,
-			compressionAlgorithm: CompressionAlgorithms.lz4,
-		},
-	},
-	fluidDataObjectType: DataObjectFactoryType.Test,
-};
+const compressionSuite = (getProvider) => {
+	describe("Compression", () => {
+		let provider: ITestObjectProvider;
+		let localMap: ISharedMap;
+		let remoteMap: ISharedMap;
+		const testContainerConfig: ITestContainerConfig = {
+			registry: [["mapKey", SharedMap.getFactory()]],
+			runtimeOptions: {
+				compressionOptions: {
+					minimumBatchSizeInBytes: 10,
+					compressionAlgorithm: CompressionAlgorithms.lz4,
+				},
+			},
+			fluidDataObjectType: DataObjectFactoryType.Test,
+		};
 
-describeFullCompat("Op Compression", (getTestObjectProvider) => {
```

Comment:
Given that we used describeFullCompat(), how did it slip through? Is it because LTS version of loader (that should be used here) was rejected for other reasons (like not having submitBatchFn), and thus we never experienced sending compressed ops with old loader?

Reply:
It's because we only look 1-2 version back (right now only 1) and it slipped. LTS will work as it won't be sending compressed ops at all.